README.md

```diff
@@ -1,19 +1,63 @@
 # HWSC-File-Transaction-SVC
 This is one of the microservices in the Humpback Whale Social Call project, which allows user to manage their files. 
-### Purpose
 
+### Purpose
 This provides file management service which includes the following:
-  - Downloading
-  - Uploading
-  - Deleting
+- Downloading
+- Uploading
+- Deleting
+
 ## Prerequisites
-- Python 3.7
+- Python 3.7 or greater
+- pip
+- pipenv
 - Webstorm
 
-## How to install Dependencies
- - TO-DO
+## Check your python version
+- On your terminal, type [python --version] to check the python version
+
+## How to install Python3.7 with Homebrew
+- If you don't have python version 3.7, you can install it with Homebrew
+- On your terminal, type [ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"] to install Homebrew
+- Insert Homebrew directory into your path by typing [sublime ~./profile]. sublime can be substituted with your favorite text editor
+- On the opening text file, insert [export PATH=/usr/local/bin:/usr/local/sbin:$PATH] as a path
+- Type [brew install python] on the command to install python 
+- Type [python3 --version] to check the version is 3.7
+- Whenever you work with python3.7, you need to call it by [python3]
+
+## How to install pipenv with Homebrew
+- If you installed python3.7 with Homebrew, you would have pip installed already
+- Use pip to install pipenv by typing [pip install --user pipenv] as a command
+
+## How to manage dependency with pipenv (to create Pipfile and Pipfile.lock)
+- Spawn a shell in a virtual environment by typing [pipenv shell]
+- If you are not already in the project directory, change into your projects directory by doing cd [project directory]
+- In created shell, you run the bash script by typing [./generate.sh]
+- This will create Pipfile and Pipfile.lock files that are used for dependency management
+- Pipfile: track which dependencies your project needs in case you need to reinstall them
+- Pipfile.lock: enables deterministic builds by specifying the exact requirements for reproducing an environment
+
+## How to manage dependency with pipenv (project already contains generated Pipfile and Pipfile.lock)
+- Type [pipenv install] on the command
+- This does installation that from Pipfile.lock
+
+## Pull the proto file and compiled pb from the hwsc-api-blocks
+- Working on it
+
+## Additional useful commands
+- pip install pytest --dev (Unit test for the application)
+- pipenv install -dev (Put dependency in [dev-packages])
+- pipenv install -env (Update Pipfile.lock)
+- pipenv lock (lock the environment so ensures the same dependency in production)
+- pipenv install --ignore-pipfile (Ignore the Pipfile for installation and use what's in the Pipfile.lock. This might be useful after you lock for the production, install the last successful environment)
 
 # How to run service
+
```

Comment:
Which pipenv command should i use before ``pipenv run python nameofthefile.py``?